libs/server/Resp/Parser/SessionParseState.cs

```diff
@@ -71,22 +67,212 @@ public void Initialize(ref ArgSlice[] buffer, int count)
             bufferPtr = (ArgSlice*)Unsafe.AsPointer(ref buffer[0]);
         }
 
+        /// <summary>
+        /// Initialize the parse state with one argument
+        /// </summary>
+        /// <param name="arg">Argument to initialize buffer with</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void InitializeWithArgument(ArgSlice arg)
+        {
+            Initialize(1);
+
+            buffer[0] = arg;
```

Comment:
Use bufferPtr, rename buffer to bufferRoot, add Slice for SPS.

Reply:
Deferring to a separate PR